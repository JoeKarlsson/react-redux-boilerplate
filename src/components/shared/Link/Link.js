import React from 'react';
import PropTypes from 'prop-types';

const STATUS = {
	HOVERED: 'hovered',
	NORMAL: 'normal',
};

class Link extends React.Component {

	constructor(props) {
		super(props);

		this._onMouseEnter = this._onMouseEnter.bind(this);
		this._onMouseLeave = this._onMouseLeave.bind(this);

		this.state = {
			class: STATUS.NORMAL,
		};
	}

	_onMouseEnter() {
		this.setState({ class: STATUS.HOVERED });
	}

	_onMouseLeave() {
		this.setState({ class: STATUS.NORMAL });
	}

	render() {
		return (
			<a
				className={this.state.class}
				href={this.props.page || '#'}
				onMouseEnter={this._onMouseEnter}
				onMouseLeave={this._onMouseLeave}
			>
				{this.props.children}
			</a>
		);
	}
}

Link.propTypes = {
	children: PropTypes.string,
	page: PropTypes.string,
};

Link.defaultProps = {
	children: '',
	page: '',
};

export default Link;
