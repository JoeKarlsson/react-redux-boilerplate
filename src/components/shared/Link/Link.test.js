import React from 'react';
import renderer from 'react-test-renderer';
import Link from './Link';

test('Link changes the class when hovered', () => {
	const component = renderer.create(
		<Link page="http://www.instagram.com">Instagram</Link>,
	);
	let tree = component.toJSON();
	expect(tree).toMatchSnapshot();

	// manually trigger the callback
	tree.props.onMouseEnter();
	// re-rendering
	tree = component.toJSON();
	expect(tree).toMatchSnapshot();

	// manually trigger the callback
	tree.props.onMouseLeave();
	// re-rendering
	tree = component.toJSON();
	expect(tree).toMatchSnapshot();
});
