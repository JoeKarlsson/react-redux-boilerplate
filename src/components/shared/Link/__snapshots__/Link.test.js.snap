// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Link changes the class when hovered 1`] = `
<a
  className="normal"
  href="http://www.instagram.com"
  onMouseEnter={[Function]}
  onMouseLeave={[Function]}
>
  Instagram
</a>
`;

exports[`Link changes the class when hovered 2`] = `
<a
  className="hovered"
  href="http://www.instagram.com"
  onMouseEnter={[Function]}
  onMouseLeave={[Function]}
>
  Instagram
</a>
`;

exports[`Link changes the class when hovered 3`] = `
<a
  className="normal"
  href="http://www.instagram.com"
  onMouseEnter={[Function]}
  onMouseLeave={[Function]}
>
  Instagram
</a>
`;
